ITR/data/data_warehouse.py

```diff
@@ -36,8 +36,13 @@ def __init__(self, company_data: CompanyDataProvider,
         self.temp_config = tempscore_config
         self.column_config = column_config
         self.company_data = company_data
-        self.company_data._calculate_target_projections(benchmark_projected_production)
-        
+        self.company_data._calculate_target_projections(benchmark_projected_production,
+                                                        benchmarks_projected_ei.scope_to_calc)
+
+        # If benchmark's scope to calculate is S3 - skip shifting data into S1S2
+        if benchmarks_projected_ei.scope_to_calc == EScope.S3:
```

Comment:
From slack chat with Michael:

Greetings!  I’ve installed and run the GUI tool in your branch and the tool appears to be producing the same results as the “production-centric” tool, which is good.  I also looked at the Jupyter Notebook, and the obvious change I see is that you overwrote the July data with the September data, rather than changing the path to the newer notebook.  That’s OK, but nowhere did I see any S3-specific calculations, observations, etc.  Can you make your S3 work visible in one or both?  And perhaps reset the default spreadsheet path to the September version?  Thanks!

Reply:
From slack chat with Michael:

I notice two things when I make changes to the notebook.  Consider the following:
~~~~
 temperature_score = TemperatureScore(
     time_frames = [ETimeFrames.LONG],
     scopes=[EScope.S1S2, EScope.S3],
     aggregation_method=PortfolioAggregationMethod.WATS # Options for the aggregation method are WATS, TETS, AOTS, MOTS, EOTS, ECOTS, and ROTS.
 )
 enhanced_portfolio = temperature_score.calculate(data_warehouse=template_provider, portfolio=companies)
~~~~
Then
~~~~
 with warnings.catch_warnings():
     warnings.simplefilter("ignore")
     display(enhanced_portfolio[['company_name', 'time_frame', 'scope', 'temperature_score']])
~~~~
Then
~~~~
 data = enhanced_portfolio[(enhanced_portfolio.scope.isin([EScope.S1S2, EScope.S3]))&(enhanced_portfolio.temperature_score>Q_(3, 'delta_degC'))]
 display(data)
~~~~
Then
~~~~
 aggregated_scores = temperature_score.aggregate_scores(enhanced_portfolio)
 print(f"Temperature Score aggregation method = {temperature_score.aggregation_method}")
~~~~
We see a lot of noise about
~~~~
 A value is trying to be set on a copy of a slice from a DataFrame.
 Try using .loc[row_indexer,col_indexer] = value instead
~~~~
 
 See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
Which is indicative of some badly behaved dataframe code done in the aggregate_scores calculation.

Reply:
From slack chat with Michael:

Moving along, consider this:
~~~~
 grouping = ['sector', 'region']
 temperature_score.grouping = grouping
 grouped_portfolio = temperature_score.calculate(data_warehouse=template_provider, portfolio=companies)
 grouped_aggregations = temperature_score.aggregate_scores(grouped_portfolio)
~~~~
Then this
~~~~
 analysis_parameters = ([ETimeFrames.LONG], [EScope.S1S2], grouping)
 plot_grouped_heatmap(grouped_aggregations, analysis_parameters)
~~~~
Then this:
~~~~
 analysis_parameters_s3 = ([ETimeFrames.LONG], [EScope.S3], grouping)
 plot_grouped_heatmap(grouped_aggregations, analysis_parameters_s3)
~~~~
The two heatmaps show identical results.  I’d expect that if we are properly segregating scopes, we’d see different contributions from isolated scopes.  What’s the overall theory of calculating based on combined scopes vs. calculating within scopes and then doing a weighted combination only when we bring them together (such as S1S2S3)?

Reply:
@MichaelTiemannOSC  All right, Michael, here is some summary:

> I also looked at the Jupyter Notebook, and the obvious change I see is that you overwrote the July data with the September data

as we discussed in slack, i didn't do it. I didn't touch Jupiter in this PR

> Can you make your S3 work visible in one or both?

i couldn't make Jupiter lab work on my PC. I got stuck with `GITHUB_TOKEN`, which i don't have currently.
Overall, Jupiter lab is so problematic for me to make work, that i would appreciate, if we could allow me to not bother about it. It doesn't seem user-friendly to me, so, maybe there is no value for our non-team users, if i make it work or not. Non-team users of our Open Source tool most probably use `ITR_UI` anyway - not everyone has a `GITHUB_TOKEN`, to proceed with Jupiter.
I managed to make `ITR_UI` work on my PC, i'd rather limit my contributions to this GUI.

> We see a lot of noise about
>
> A value is trying to be set on a copy of a slice from a DataFrame.
> Try using .loc[row_indexer,col_indexer] = value instead

> Which is indicative of some badly behaved dataframe code done in the aggregate_scores calculation.

i will double-check, what in my changes caused it

> The two heatmaps show identical results. I’d expect that if we are properly segregating scopes, we’d see different contributions from isolated scopes.

That's how my PR works:
scope to calculate is based on EI benchmark.
if EI benchmark, which we select in `ITR_UI`, has S1S2 - it will be calculated.
Otherwise, if EI benchmark has S3 - it will be calculated.
Otherwise, there is no scope to calculate

> What’s the overall theory of calculating based on combined scopes
> ...
> (such as S1S2S3)?

i didn't touch this aspect in my "S3-only" task.
all i do here - is either calculate all for S1S2, or calculate all for S3. I didn't touch S1S2S3

Reply:
>> Which is indicative of some badly behaved dataframe code done in the aggregate_scores calculation.

> i will double-check, what in my changes caused it

i pushed a commit, which fixes these warnings.
from my perspective, i worked on all your comments.
looking forward to your feedback

Reply:
> Can you make your S3 work visible in one or both?

Update: thanks to the help of @MichaelTiemannOSC, i was able to run Jupiter notebook on this branch.
Yes, agree, I will make `EScope.S3` visible in the notebook "quick_template_score_calc.ipynb"

Reply:
... and done: i pushed 2 Jupier Notebook commits:

* Removed unused github token
* Add S3 to calculated scores